site/en/blog/chrome-ux-report-history-api/index.md

```diff
@@ -0,0 +1,191 @@
+ --header 'Content-Type: application/json' \
+ --data '{"url": "https://web.dev/blog/"}'
+```
+
+Page-level (and origin-level) historical data is subject to [the same eligibility requirements](/docs/crux/methodology/#eligibility) as the rest of CrUX, and so pages in particular may not have a complete historical record. In these cases the "missing" data will be represented by `"NaN"` for the `histogramTimeseries` densities and `null` for the `percentilesTimeseries`. The reason for the difference is the histogram densities are always numbers, while the percentiles can be numbers or strings (CLS uses strings, even if they look like numbers).
+
+## Visualizing the data
+
+So, you may ask, why is the data shaped in this way? It was found that this makes it easy to plot graphs. For example, here is a graph for the p75 values for [Interaction To Next Paint (INP)](https://web.dev/inp/) for [https://web.dev](https://web.dev):
+
+{% Img src="image/kd0zRXNeKScEzrwMmhY1NNvIzeM2/fgLRTt2WB2Yd67bL3mMc.png", alt="Time series graph of p75 value showing a regression around November 2022", width="512", height="339" %}
+
+In this line chart, each value on the y-axis is a p75 value from the `p75s` time series, and the x-axis is the time, which is set up as the `lastDate` for each collection period.
+
+Here is a graph for the histogram bin time series—known as a tri-bin chart—because these histograms have three bins.
+
+{% Img src="image/kd0zRXNeKScEzrwMmhY1NNvIzeM2/3izH0o8vNvZqXGMWYCFJ.png", alt="Stacked bar chart showing how the relative proportions of good, needs improvement and poor changes over time.", width="678", height="353" %}
+
+The x-axis is set up as the `lastDate` for each collection period. This time, however, the y-axis is the percentage of page loads falling into a specific range for the INP metric, shown through a stacked bar chart. The p75 chart provides a quick overview, and within a single chart, it’s relatively easy to add multiple metrics, or show lines for both `PHONE` and `DESKTOP`. The tri-bin chart gives a sense of the distribution of metric values measured during each collection period.
+
+For example, even though the p75 chart suggests that [https://web.dev](https://web.dev) had almost acceptable INP values during the observation period, the tri-bin chart shows that for a small fraction of page loads, the INP was actually poor. In both charts, it’s relatively easy to infer that there was a performance regression which started toward the end of October, and was fixed by mid-November.
+
+To generate such charts yourself, we have created an example [Colab](https://colab.research.google.com/). A Colab, or 'Colaboratory', allows you to write and execute Python in your browser. The [CrUX History API Colab](https://colab.research.google.com/github/GoogleChrome/CrUX/blob/main/colab/crux-history-api.ipynb) [[source]](https://github.com/GoogleChrome/CrUX/blob/main/colab/crux-history-api.ipynb) uses Python to make calls to the API and chart the data.
```

Comment:
in your browser -> from within your browser (the python runs in some cloud process)